src/App.tsx

```diff
@@ -1,27 +1,48 @@
-import React from 'react';
-import './App.scss';
+import React, { useState } from 'react';
+import { getAll, get5First, getRedGoods } from './api/goods';
+
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { Good } from './types/Good';
+
+import './App.scss';
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+
+  const handleDownloadedGoods = (loadedGoods: Good[]): void => {
+    setGoods(loadedGoods);
+  };
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => getAll().then(handleDownloadedGoods)}
+      >
+        Load all goods
+      </button>
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={() => get5First().then(handleDownloadedGoods)}
```

Comment:
handleDownloadedGoods function seems to be redundant, it does nothing except setting goods. You can use just setGoods